extra_docs/jfs/README.md

```diff
@@ -0,0 +1,194 @@
+```typescript
+jfs.fileset('my_awesome_fileset').upload([
+  record1, record2, ...
+]).subscribe(fileRecord => {
+   console.log(fileRecord);
+   /* output:
+   { id: '11a12f17-8367-4114-a588-ae98a6cb3cda',
+      created_at: '2019-05-24T07:17:37.325882Z',
+      updated_at: '2019-05-24T07:17:37.325882Z',
+      name: null,
+      size: null,
+      type: null,
+      url: null,
+      status: 'in_progress' }
+    */
+});
+```
+
+### [File upload status](#status)
+After you upload a file you will receive file status. Status displays the progress of file uploading. 
+Possible values are: 
+
+- `in_progress` 
+- `completed`
+- `failed` 
+
+ The first status you will get in response from `upload()` is `in_progress` (if you did not set true to the 
+`subscribeForTheFileUploading` config property). It means that file record has been created and currently file is 
+uploading to the persistent storage (whether AWS bucket or something else). 
+There are two ways to track further file status changes - you can subscribe to the fileset changes with 
+`realTimeModule` or you can set `subscribeForTheFileUploading` config option and SDK subscribes to that 
+status automatically.
+
+Use `realTimeModule`:
+```typescript
+import { jexiaClient, fileOperations, realTime } from 'jexia-sdk-js';
+
+const jfs = fileOperations();
+
+jexiaClient().init({
+  projectID: 'your-project-id',
+  key: 'your-project-key',
+  secret: 'your-project-secret'
+}, jfs, realTime());
+
+const fileset = jfs.fileset('my_awesome_fileset');
+
+fileset.upload(record).pipe(
+  concatMap(fileRecord => fileset.watch().pipe(
+    filter(event => event.data[0].id === fileRecord.id),
+    takeWhile(event => event.data[0].status !== 'completed', true),
+    map(event => {
+      fileRecord.status = event.data[0].status;
+      return fileRecord;
+    })
+  ))
+).subscribe();
+```
+As you can see, there are some details you should take care of: filtering events by record id (`watch()` returns 
+all events from the fileset), unsubscribing from watch observable with `takeWhile`. Better to use the next option.
+
+Use `subscribeForTheFileUploading` config property (notice that you still need to activate realTimeModule):
+```typescript
+import { jexiaClient, fileOperations, realTime } from 'jexia-sdk-js';
+
+const jfs = fileOperations({
+  subscribeForTheFileUploading: true    
+});
+
+jexiaClient().init({
+  projectID: 'your-project-id',
+  key: 'your-project-key',
+  secret: 'your-project-secret'
+}, jfs, realTime());
+
+const fileset = jfs.fileset('my_awesome_fileset');
+
+fileset.upload(record).subscribe(fileRecord => {
+  console.log(fileRecord);
+  /* output:
+    { id: '11a12f17-8367-4114-a588-ae98a6cb3cda',
+      created_at: '2019-05-24T07:17:37.325882Z',
+      updated_at: '2019-05-24T07:18:40.455764Z',
+      name: 'logo.png',
+      size: 42341,
+      type: 'png',
+      url: 'https://...',
+      status: 'completed' }
+  */
+});
+```
+
+### [Fileset CRUD operations](#crud)
```

Comment:
I think we do not have Create operator. Update + RUD?

Reply:
we have `insert` which is basically CREATE